10up/classifai
# Feature/issue 368 auto updates

### Description of the Change

This PR implements automatic plugin updates for sites with active licenses. Instead of requiring site managers to manually download & upload a new copy of ClassifAI, this will utilize WP Core's built-in plugin updater, allowing users with a valid license to install the update directly through the wp-admin Plugins page.

**NOTE:** If the site does not contain a valid license, the plugin will not indicate an update is available. A valid license must be available for the updater to initialize.

**Additional Changes:**
- ~~`package-lock.json` has been rebuilt to `lockfileVersion: 2` to support the latest Node/NPM versions.~~
- Minor formatting updates.


**Closes Issue:** https://github.com/10up/classifai/issues/368

### How to test the Change

1. Pull down the associated branch [`feature/issue-368-auto-updates`](https://github.com/10up/classifai/tree/feature/issue-368-auto-updates)
2. Change the plugin version via the Header comment to a lower version than the current:
    - https://github.com/10up/classifai/blob/feature/issue-368-auto-updates/classifai.php#L7
3. Login to wp-admin on a site WITHOUT a valid license.
    - Plugin should **not** indicate an update is available
4. Login to wp-admin on a site WITH a valid license
    - Alternatively change the [`\Classifai\Admin\Updater::license_check()`](https://github.com/10up/classifai/blob/feature/issue-368-auto-updates/includes/Classifai/Admin/Update.php#L96-L107) method to return `true`.
5. Visit Plugins page to view the available updates.
    - ClassifAI should indicate a new version is available for download
6. Click "Update Plugin" to pull down and install the latest plugin version.
    - DO NOT commit your version # changes or after plugin update.
 

### Changelog Entry

> Added - Support for WordPress auto updates for sites with a valid ClassifAI registration key.

### Credits

Props @TylerB24890 

### Checklist:
